### DISCUSSION TITLE
Support Resources for Developers

### DISCUSSION THREAD

**Discussion by kathrynwp** [WordPress/developer-blog-content#54] [Topic Ideas]
Some folks might think that developers know everything, having all the WordPress knowledge and never needing help themselves. However, that is not the case! 

Let's create a roundup of resources to assist developers in getting help when they need it. This could cover angles like:

- I want opinions on a technical approach to building my plugin
- I'm building a block theme for the first time but I'm not sure what the requirements are for submitting it to the WordPress.org directory
- I'm supporting a plugin in the WP.org forums but am not sure how to handle out-of-scope questions
- I'm interested in contributing to Core but not sure where to get started
- I have some questions about accessibility in a plugin I'm developing
- I'm building a custom block and have run into a roadblock
- I've run into a bug with something and am not sure where to report it (i.e. core, block editor, theme, plugin, forum)

What do folks think? Should we narrow down the focus somehow or consider more varied scenarios?

Are there other angles or ways to approach this?

**Comment by bph** [comment]
So from the fact that you started this discussion, I conclude that those questions are not answered via the Support Forums and that there is the need to create Resources on the Developer Blog. Below I am just very briefly try to outline each case, and see what I, personally, know and what resources are available of the top of my head. This might help to add scope to this resource request and maybe later decide the number of pages or grouping. 

### I want opinions on a technical approach to building my plugin 

I don't think there is a place for that on .org site. There are membership sites, that might help out with consulting for plugins developers. Might be a discussion in various groups on the social web or on Reddit space? Or even Stackexchange? There are a group of books available regarding plugin and app development. A link to the plugin handbook might also get them started, but it feels like they are already past that point. 

### I'm building a block theme for the first time, but I'm not sure what the requirements are for submitting it to the WordPress.org directory 

For block themes the same guidelines apply as for classic themes. It could be answered with links to Theme Handbook on guidelines and how to submit. And even connect with members of the team in the #themereview channel. 

### I'm supporting a plugin in the WP.org forums but am not sure how to handle out-of-scope questions

Plugin Handbook, guidelines, and maybe answer out-of-scope questions anyway? We are all one community<g> 
Join the #forums slack and ask (nicely) for advice. Don't know what advice the moderators give. They have opinions for sure. 

### I'm interested in contributing to Core but not sure where to get started

Here we have a myriad of resources at Make.wordPress.org/core
- Contributor handbook on Core
- Contributor handbook on Gutenberg
- Meeting schedule core/core-editor/core-js etc. 
- Meeting schedule New contributors
- Meeting schedule Triage 
- How to work with the community page. 

### I have some questions about accessibility in a plugin I'm developing

- WordPressTV has a few recording on talks from accessibility experts in the space
- Maybe accessibility testing plugins and test sites? 
- Do we know accessibility clearing house for consultants? 
- If it's within the scope of the team #accessibility channel and ask?
- There is a blog post idea in how to use the built-in features of WordPress (components, scripts, and tools) for a 3rd party plugin. Then they are at least as accessible as WordPress is and someone else is maintaining the components

### I'm building a custom block and have run into a roadblock

- Come to Learn WordPress online learning events about your topic. 
- Look through the developer Blog for a topic that might fit your problem,
- There are frequent Developer Hours, and AMA's scheduled. 
- Post to StackExchange or 
- Discussion on the Gutenberg repo

### I've run into a bug with something and am not sure where to report it (i.e., core, block editor, theme, plugin, forum)

- Troubleshooting WordPress 
- More discovery necessary: Could be a tutorial: "the troubleshooting toolbox for the beginner developer"
- How to use the Site health and troubleshooting plugin for 
- How to find your log files
- How to use debugging. 
- Browser Developer tools.  

@kathrynwp Thanks for submitting this fabulous list of common support cases that don't seem to have good coverage or are hard for outside developers to find. Maybe you could also check if there is something missing, or I misunderstood the angle. 

Anyone has additional ideas how to tackle this?

**Comment by juanmaguitar** [comment]
Thanks, @kathrynwp, and @bph, for shaping this idea.
The angle Kathryn provided, and the resources Birgit provided are excellent ways to shape this content. 
The goal I see for a post like this is to highlight **resources (handled mainly by the community of WordPress), where developers can ask technical questions (and get answers).**

So, taking into account all your feedback, I think a good angle for this post would be:
- Using the strategy of questions (use cases) suggested by @kathrynwp
- Using the feedback provided by @bph as guidance for answering those questions
- Highlight resources available in the community where people can ask questions (or open conversations) and get answers from the community

---

Some examples of resources that I think are a bit hidden and could be highlighted by a post like this, are:
- [Developing with WordPress Forum](https://wordpress.org/support/forum/wp-advanced/) → This a good resource [where developers can ask technical questions](https://wordpress.org/support/topic/using-the-developing-with-wordpress-sub-forum/) and get support from the community. Some examples
  - _[How do I hook into content after it is completely compiled](https://wordpress.org/support/topic/how-do-i-hook-into-content-after-it-is-completely-compiled/)_
  - _[A way to include product comparison tables within mass CSV product import?](https://wordpress.org/support/topic/comparison-tables-within-mass-csv-product-import/)_
- https://wordpress.stackexchange.com/ → This resource is mentioned in the [#core-editor](https://wordpress.slack.com/archives/C02QB2JS7) Slack channel so I think it's worth mentioning it as another resource where developers can ask technical questions about WordPress
- https://github.com/WordPress/gutenberg/discussions → This resource is excellent for opening technical conversations to understand Gutenberg better or the technical decisions around it. This is also a great resource to share the experience of developing with it.
- New Contributor meetings on the 2nd and 4th Wednesdays at 19:00 UTC. Upcoming WordPress Meetings: https://make.wordpress.org/meetings/  → For developers that want to contribute but don't know where to start and would like some guidance about it

**Comment by kathrynwp** [comment]
Thank you for the feedback, @bph and @juanmaguitar!

I will start working on a draft taking into account the input above, and will share it when it's ready. I'll try to limit the scope a bit so that it isn't too long or too broad, and we'll see how that goes. I'll keep you posted here.

**Comment by ndiego** [comment]
This is an excellent topic and something that WordPress developers definitely need. However, I am not sure that a blog post is necessarily the _best_ format for this. A blog post with a narrower scope, so it's manageable, would be great. But I would love to see an evergreen page, or a few different pages created that are dedicated to developer resources. These pages could be updated routinely and could even be referenced in blog posts. 🤔

**Comment by kathrynwp** [comment]
Hey there, just a heads-up that I'm going to be stepping back from this one but anyone is of course welcome to take over shepherding this along, in whatever form it may eventually take! Thanks to all for their input so far.

**Comment by Grzegorz71246** [comment]
On our website, we have a special "Dev Tips and Tricks" section where we publish articles with code and sometimes even links to Git repos for some WP ideas or solutions. Here are some topics we've covered, just to give you an idea:

- reCAPTCHA v3 – WordPress implementation
- Migrate learnDash version 2 to 4
- How to create a WP plugin with admin forms
- WP admin custom forms – how to display errors?
- How to translate WooCommerce
- WooCommerce orders report sent by e-mail

and many more. Users can comment freely and ask us questions if they have any. We publish some of our WP solutions on Reddit too.

Here is a link to the relevant section on our website: https://www.createit.com/blog/tags/wordpress/
We cover a lot of webdev topics, but WP remains at the top of our list, since this is our favorite CMS and we've been working with it for almost as long as it exists.

**Reply by justintadlock** [reply]
Most of those are related to third-party plugins, so they wouldn't be suitable for the Dev Blog. However, for those that are not (like the admin forms ones), it could be worth exploring as separate topics. I encourage reading the [how to contribute](https://developer.wordpress.org/news/how-to-contribute/) page and submit them as their own topics.

**Comment by bph** [comment]
As the discussion didn't seem to progress towards a concrete topic idea, I closed the discussion.
